core/src/main/java/io/grpc/internal/JsonParser.java

```diff
@@ -76,11 +79,13 @@ private static Object parseRecursive(JsonReader jr) throws IOException {
     }
   }
 
-  private static Map<String, ?> parseJsonObject(JsonReader jr) throws IOException {
+  private static Map<String, ?> parseJsonObject(JsonReader jr)
+      throws IOException {
     jr.beginObject();
     Map<String, Object> obj = new LinkedHashMap<>();
     while (jr.hasNext()) {
       String name = jr.nextName();
+      checkArgument(!obj.containsKey(name), "Duplicate key found: " + name);
```

Comment:
This allocates the error string even when there isn't an error. Use built-in formatting available to checkArgument.